gdnative-core/src/core_types/mod.rs

```diff
@@ -1,11 +1,16 @@
-//! Types that represent core-datatypes of Godot.
+//! Types that represent core data types of Godot.
```

Comment:
Just confirming, most of the changes in here are for organizational purposes, correct? If so, looks good!

Reply:
Yes, I don't want to introduce any semantic changes in this PR.
There might be breaking changes due to new module structure, but functionality-wise, we should not lose anything.